lib/error-helper.js

```diff
@@ -18,6 +23,26 @@ const checkErrorResponse = function(badgeData, err, res, notFoundMessage = 'not
   }
 };
 
+checkErrorResponse.asPromise = function ({ notFoundMessage }) {
+  return async function ({ buffer, res }) {
+    if (res.statusCode === 404) {
+      throw new NotFound(notFoundMessage);
+    } else if (res.statusCode !== 200) {
+      throw new InvalidResponse();
+    }
+    return { buffer, res };
```

Comment:
I like the way this just transparently passes through the inputs so it 'chains' nicely